app/controllers/nonprofits_controller.rb

```diff
@@ -55,10 +55,18 @@ def create
   end
 
   def update
-    flash[:notice] = "Update successful!"
-    current_nonprofit.update_attributes params[:nonprofit].except(:verification_status)
-    current_nonprofit.clear_cache
-    json_saved current_nonprofit
+    @form = NonprofitSettingsForm.new(
+      nonprofit: current_nonprofit,
+      attributes: params[:nonprofit].except(:verification_status)
+    )
+
+    if @form.save
+      flash[:notice] = "Update successful!"
+      current_nonprofit.clear_cache
+      json_saved current_nonprofit
+    else
+      json_saved current_nonprofit, 500
+    end
```

Comment:
Just realized: we need to check if the `current_user` saving the nonprofit is an nonprofit_admin or a super_admin. A nonprofit_associate has the power to modify the `Nonprofit` generally but they can't modify the `require_two_factor` attribute.

Reply:
The submit button is hidden in the view for associates, but agreed, safer to check here too. I'll update it.